src/main.rs

```diff
@@ -30,8 +30,16 @@ use crate::util::deserialize_file;
 use crate::widgets::text::TextWidget;
 use crate::widgets::{I3BarWidget, State};
 
-#[tokio::main(flavor = "current_thread")]
-async fn main() {
+fn main() {
+    tokio::runtime::Builder::new_current_thread()
+        .max_blocking_threads(2)
```

Comment:
I'm not sure if this number should be bigger, but with `max_blocking_threads(2)` bar has no freezes.